core/src/main/java/org/javarosa/form/api/FormEntrySessionReplayer.java

```diff
@@ -0,0 +1,89 @@
+package org.javarosa.form.api;
+
+import org.javarosa.core.model.FormIndex;
+import org.javarosa.core.model.data.AnswerDataFactory;
+import org.javarosa.core.model.data.IAnswerData;
+import org.javarosa.core.model.data.UncastData;
+
+/**
+ * Replay form entry session. Steps through form, applying answers from the
+ * session to corresponding questions by matching form indices. Replay aborts
+ * if form indices don't match and stops when the session is empty.
+ *
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public class FormEntrySessionReplayer {
+    private final FormEntrySession formEntrySession;
+    private final FormEntryController formEntryController;
+
+    private FormEntrySessionReplayer(FormEntryController formEntryController,
+                                     FormEntrySession formEntrySession) {
+        this.formEntryController = formEntryController;
+        this.formEntrySession = formEntrySession;
+    }
+
+    public static void tryReplayingFormEntry(FormEntryController formEntryController,
+                                             FormEntrySession formEntrySession) {
+        FormEntrySessionReplayer replayer =
+                new FormEntrySessionReplayer(formEntryController, formEntrySession);
+        if (replayer.isRestoringFormSession()) {
+            replayer.replayForm();
+        }
+    }
+
+    private boolean isRestoringFormSession() {
```

Comment:
Nitpicking, but method name here seems weird. What you're checking is more like `hasFormSessionToReplay`, right?